### ISSUE TITLE
MPI Error

### ISSUE THREAD

**Issue by etamsylate-pupu** [senp98/wdm#1]
Thanks so much for the code, it helped me a lot.
But I'm having some problems running it.
`  File "...../wdm/wdm/dist_util.py", line 10, in <module>
    from mpi4py import MPI
ImportError: libmpi.so.12: cannot open shared object file: No such file or directory
`
I have installed the related package and openmpi-4.1.5, by the way, I run it on CentOS 

![mamba_list](https://github.com/senp98/wdm/assets/57471206/8cf60727-40e6-40c2-b527-d1456f025652)

**Comment by etamsylate-pupu** [comment]
Do you have this problem? Thx for your help

**Comment by senp98** [comment]
Thank you. We conducted all experiments on Ubuntu 22.04 and we didn't have this problem... I think it's an environment issue related to the mpi4py package on CentOS.

**Comment by etamsylate-pupu** [comment]
It might be a version error of openmpi, the openmpi-4.1.5 corresponds to libmpi.so.40, not libmpi.so.12.
But before I successfully installed the lower version of openmpi, I reinstalled miniforge in a normal user env, and this time the installed mpi4py and mpich were not the same as the previously installed version, and ran successfully without installing openmpi. I don't understand what's going on...but it doesn't matter.
![image](https://github.com/senp98/wdm/assets/57471206/de18589a-4970-4a64-9a85-b9d362712864)
Thx very much for your help!
